content/en/blog/_posts/2022-12-16-non-graceful-node-shutdown-to-beta.md

```diff
@@ -0,0 +1,80 @@
+
+## What is a Non-Graceful Node Shutdown
+
+A Node Shutdown can be `Graceful` only if kubelet's Node Shutdown Manager can detect the node shutdown action. However, there are cases where kubelet's Node Shutdown Manager may not detect a node shutdown action. This could happen because the shutdown command does not trigger the [Inhibitor Locks](https://www.freedesktop.org/wiki/Software/systemd/inhibit) mechanism used by kubelet or because of a user error, i.e., the `ShutdownGracePeriod` and `ShutdownGracePeriodCriticalPods` are not configured correctly.
+
+When a node is shut down but not detected by kubelet's Node Shutdown Manager, it becomes a non-graceful node shutdown. Non-graceful node shutdown is a problem for stateful apps. If a node containing a pod that is part of a StatefulSet is shut down in a non-graceful way, the pod will be stuck in `Terminating` status and cannot move to a new running node. Similarly, pods that ReplicaSets create as part of a Deployment will be stuck in `Terminating` status on the shutdown node forever. However, they will come up on another running node and be stuck in the `ContainerCreating` status on the new node. As a result, the application will not function properly. If the original shutdown node comes up, the old pod will be deleted by kubelet, and the new pod will be created successfully on a different running node.
+
+## What’s new in Beta
+
+With the promotion of the Non-Graceful Node Shutdown feature to beta, the feature gate `NodeOutOfServiceVolumeDetach` is enabled by default on `kube-controller-manager` instead of being opt-in.
+
+New metrics `force_delete_pods_total`, number of pods that are being forcefully deleted since the Pod GC Controller started, and `force_delete_pod_errors_total`, number of errors encountered when forcefully deleting the pods since the Pod GC Controller started, are also added in the Pod GC Controller.
+
+## How does it work
+
+In the case of a node shutdown, if a graceful shutdown is not working or the node is in a non-recoverable state due to hardware failure or broken OS, the user can manually add an `out-of-service` taint on the Node. For example, this can be `node.kubernetes.io/out-of-service=nodeshutdown:NoExecute` or `node.kubernetes.io/out-of-service=nodeshutdown:NoSchedule`. This will trigger pods on the node to be forcefully deleted if there are no matching tolerations on the pods.  Persistent volumes attached to the shutdown node will be detached, and new pods will be created successfully on a different running node.
+
+```
+kubectl taint nodes <node-name> node.kubernetes.io/out-of-service=nodeshutdown:NoExecute
+```
+
+Note: Before applying the out-of-service taint, you must verify that a node is already in shutdown or power-off state (not in the middle of restarting), either because the user intentionally shut it down or the node is down due to hardware failures, OS issues, etc.
```

Comment:
```suggestion
**Note:** Before applying the out-of-service taint, you must verify that a node is already in shutdown
or power-off state (not in the middle of restarting), either because the user intentionally shut it down
or the node is down due to hardware failures, OS issues, etc.
```